packages/iTwinUI-react/src/core/ButtonGroup/ButtonGroup.test.tsx

```diff
@@ -79,6 +79,47 @@ it.each(['start', 'end'] as const)(
   },
 );
 
+it.each(['start', 'end'] as const)(
+  'should handle overflow when available space is smaller than one element (%s)',
+  (overflowPlacement) => {
+    const scrollWidthSpy = jest
+      .spyOn(HTMLElement.prototype, 'scrollWidth', 'get')
+      .mockReturnValue(200);
+    const offsetWidthSpy = jest
+      .spyOn(HTMLElement.prototype, 'offsetWidth', 'get')
+      .mockReturnValue(50);
```

Comment:
When trying to replicate the issue, I found that the `scrollWidth` and `offsetWidth` were `50` and `0` respectively.

![image](https://user-images.githubusercontent.com/45748283/190423716-9613c9d2-4927-4ca5-ac3c-7c9a755ff27f.png)

I did this by adding ``console.log(`offsetWidth: ${containerRef.current.offsetWidth}`,`scrollWidth: ${containerRef.current.scrollWidth}`);`` below [line 72 of useOverflow.tsx](https://github.com/iTwin/iTwinUI-react/blob/main/packages/iTwinUI-react/src/core/utils/hooks/useOverflow.tsx#L72).

Maybe I misunderstood where these `scrollWidth` and `offsetWidth` values are being used. Are they being used in `useOverflow` for the `containerRef` (`iui-button-group`)? Or are these values being mocked for something else?

Reply:
I've been playing with it and I don't think it works at the same location in `useOverflow`, or `offsetWidth = 150` and `scrollWidth = 150` would show 3 button elements.

Reply:
Ohh, I see. I'm not too sure. Maybe I misundertood where those mocks for `scrollWidth` and `offsetWidth` are being used. So maybe they are used in some other place than the place where I put the comment. If that is the case, maybe your mock values are correct. @bentleyvk What do you think?

Reply:
`scrollWidth` and `offsetWidth` are used in `useOverflow` hook. `offsetWidth` is width of available space and `scrollWidth` is width of the space that is needed for all visible buttons to fit. So if `scrollWidth` is `200px` that means that each button's width is `200 / 3 = 66px` and `66 > 50` so no buttons fit here. Not sure if I answered your question.

Reply:
Ohh, I see. I'm slightly confused. But wouldn't all the buttons be of width of `50px` (Since the `offsetWidth` for even the button children is `50px`)? I'm slightly confused why would the buttons change its width according to the available space (`scrollWidth of parent container / number of buttons`).

Also, I'm a bit confused that if no buttons would fit here, then why is the test on [line 123 of this PR](https://github.com/iTwin/iTwinUI-react/pull/827/files#diff-bda7aa7aeee0828d62bfdcba442079e287ab78c173fc5037015a8d3e6913cec5R123) expecting 1 button?

I might be totally wrong, so I guess I'll ask you these questions later to better understand how mocks work, in order to not hold up this PR.

Reply:
Yeah, I'm still having trouble with this too.